### PR TITLE
Remove suffix #id\d{1} as part of openjdk testcase name

### LINKED ISSUE #4385: "Rerun failed jdk test cases in Grinder with jdk_custom target" didn't set BUILD_LIST
In the build https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_sanity.openjdk_s390x_linux_OpenJDK11/66/
The link for [Rerun failed jdk test cases in Grinder with jdk_custom target](https://openj9-jenkins.osuosl.org/job/Grinder/parambuild/?SDK_RESOURCE=customized&TARGET=jdk_custom&TEST_FLAG=&UPSTREAM_TEST_JOB_NAME=&DOCKER_REQUIRED=false&ACTIVE_NODE_TIMEOUT=&VENDOR_TEST_DIRS=&EXTRA_DOCKER_ARGS=&TKG_OWNER_BRANCH=adoptium%3Amaster&OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&PLATFORM=s390x_linux&GENERATE_JOBS=true&KEEP_REPORTDIR=true&PERSONAL_BUILD=false&DOCKER_REGISTRY_DIR=&ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&RERUN_ITERATIONS=0&SETUP_JCK_RUN=false&DOCKER_REGISTRY_URL_CREDENTIAL_ID=&LABEL=&EXTRA_OPTIONS=&BUILD_IDENTIFIER=OpenJDK11-Acceptance&CUSTOMIZED_SDK_URL=https%3A%2F%2Fopenj9-artifactory.osuosl.org%2Fartifactory%2Fci-openj9%2FBuild_JDK11_s390x_linux_OpenJDK11%2F70%2FOpenJ9-JDK11-s390x_linux-20230301-223813.tar.gz+https%3A%2F%2Fopenj9-artifactory.osuosl.org%2Fartifactory%2Fci-openj9%2FBuild_JDK11_s390x_linux_OpenJDK11%2F70%2Ftest-images.tar.gz&JENKINS_KEY=&ADOPTOPENJDK_BRANCH=master&LIGHT_WEIGHT_CHECKOUT=true&USE_JRE=false&ARTIFACTORY_SERVER=ci-eclipse-openj9&KEEP_WORKSPACE=false&USER_CREDENTIALS_ID=&JDK_VERSION=11&DOCKER_REGISTRY_URL=&ITERATIONS=1&VENDOR_TEST_REPOS=&JDK_REPO=https%3A%2F%2Fgithub.com%2Fibmruntimes%2Fopenj9-openjdk-jdk11.git&JCK_GIT_BRANCH=master&OPENJ9_BRANCH=master&OPENJ9_SHA=6cddd76145cf68473564905c2728d37d5ac0e016&JCK_GIT_REPO=&VENDOR_TEST_BRANCHES=&OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&UPSTREAM_JOB_NAME=&CLOUD_PROVIDER=&CUSTOM_TARGET=java/lang/ProcessBuilder/Basic.java#id0&VENDOR_TEST_SHAS=&JDK_BRANCH=openj9-staging&LABEL_ADDITION=%21sw.os.rhel.8&ARTIFACTORY_REPO=ci-openj9&ARTIFACTORY_ROOT_DIR=Test&UPSTREAM_TEST_JOB_NUMBER=&DOCKERIMAGE_TAG=&JDK_IMPL=openj9&TEST_TIME=&SSH_AGENT_CREDENTIAL=&AUTO_DETECT=true&SLACK_CHANNEL=&DYNAMIC_COMPILE=false&RELATED_NODES=&ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&APPLICATION_OPTIONS=&CUSTOMIZED_SDK_URL_CREDENTIAL_ID=e047f50a-26c4-4b82-8525-6da17ec68450&ARCHIVE_TEST_RESULTS=false&NUM_MACHINES=&OPENJDK_SHA=&TRSS_URL=&USE_TESTENV_PROPERTIES=false&BUILD_LIST=openjdk&UPSTREAM_JOB_NUMBER=&STF_OWNER_BRANCH=adoptium%3Amaster&TIME_LIMIT=10&JVM_OPTIONS=&PARALLEL=None) 
doesn't set BUILD_LIST properly. Although it does contain `BUILD_LIST=openjdk`, when you open the link the BUILD_LIST is set to functional.

### PR DESCRIPTION
**Pull request by sophia-guo**
- Jtreg doesn't support to specify testcase name suffixed with #id\d{1}
- the special character in url without encoding will also lead to wrong behaviour

Fix #4385 

Signed-off-by: Sophia Guo <sophia.gwf@gmail.com>

### CODE REVIEW

**Review by karianna — CHANGES_REQUESTED** [review]
Could do with a comment explaining why

**Comment by sophia-guo** [comment]
> Jtreg doesn't support to specify testcase name suffixed with #id\d{1}

One way to run openjdk test using JTreg is to specify testcase name, for example: java/lang/String/StringRepeat.java. When specify as java/lang/String/StringRepeat.java#id1 it won't work.

**Review by llxia — APPROVED** [review]
lgtm

File: buildenv/jenkins/JenkinsfileBase [hunk 1]
```diff
@@ -1102,6 +1102,8 @@ def addFailedTestsGrinderLink(paths=""){
 								failedTestCasesInfo = failedTestCasesInfo.substring(failedTestCasesInfo.indexOf('TEST:'))
 								failedTestCasesInfo = failedTestCasesInfo.substring(0, failedTestCasesInfo.indexOf('Test results:'))
 								failedTestCasesInfo = failedTestCasesInfo.split("\\n").join(" ").replaceAll("TEST: ", "")
+								//Remove #id suffixed to the openjdk testcase name. Jtreg doesn't work with it.
```
Comment (karianna) [inline buildenv/jenkins/JenkinsfileBase:1105]:
```suggestion
								// Remove #id suffixed to the openjdk testcase name. Jtreg doesn't work with it.
```

**Review by karianna — APPROVED** [review]

**Pull request merged**